compiler/passes/convert-typed-uast.cpp

```diff
@@ -2803,28 +2809,40 @@ TConverter::defaultValueForType(const types::Type* t,
 }
 
 Symbol*
-TConverter::makeNewTemp(const types::QualifiedType& qt, bool insertDef) {
+TConverter::makeNewTemp(::Qualifier qual, Type* t, bool insertDef) {
   auto ret = newTemp();
   ret->addFlag(FLAG_EXPR_TEMP);
-  ret->qual = convertQualifier(qt.kind());
-  ret->type = convertType(qt.type());
+  ret->qual = qual;
+  ret->type = t;
 
   if (insertDef) insertStmt(new DefExpr(ret));
 
   return ret;
 }
 
+Symbol*
+TConverter::makeNewTemp(const types::QualifiedType& qt, bool insertDef) {
+  return makeNewTemp(convertQualifier(qt.kind()), convertType(qt.type()), insertDef);
+}
+
 SymExpr*
 TConverter::storeInTempIfNeeded(Expr* e, const types::QualifiedType& qt) {
   if (SymExpr* se = toSymExpr(e)) {
     return se;
   }
 
-  // Prevent default-constructed 'QualifiedType' from slipping in.
-  INT_ASSERT(qt.hasTypePtr());
+  Symbol* t = nullptr;
+
+  if (qt.isUnknown()) {
+    t = makeNewTemp(e->qualType().getQual(), e->typeInfo());
```

Comment:
The problem with this is that there are at least some paths through `->qualType()` which seem to invoke the old resolver which is why I decided to only allow temps created with the `chpl::types::QualifiedType`. If we could figure out some way to translate without invoking the old resolver this could be done.

E.g., I know there are some primitives where the types are determined via the old resolver. I have not done due diligence but there could be more.

Maybe we hold off on adding this sort of thing if we can and if we do make sure via sanity checks/crashes that it can't actually invoke the old resolver.

Reply:
I was unaware that this was the case, so I'll look into using QualifiedType.

Reply:
Cool, thanks, maybe consider adding a comment or something so that people in the future can catch on? I thought I added one but it was probably in a bad spot or too long winded.